### ISSUE TITLE
General/Settings Site timezone drop down needs an accessible name

### ISSUE THREAD

**Issue by youbrokesomething** [Automattic/wp-calypso#75438]
### Quick summary

The drop-down selector for 'Site timezone' on General/Settings has no accessible name and can be missed by screen readers.

This is also a violation of WCAG 2A and must be fixed before we can achieve A-level compliance.

### Steps to reproduce

1. Start on Settings/General
2. Inspect the Site timezone drop-down selector `<select class="form-select">`
3. Confirm the drop-down does not have an a11y label or any accessible name for screen readers or other assistive technology.

Alternatively, use a developer tool like axe DevTools to inspect the Settings/General screen for a11y rule violations.

### What you expected to happen

The drop-down selector for Site timezone should have an accessible name that screen readers can access.

### What actually happened

The Site timezone drop-down selector `<select class="form-select">` should have an accessible name and the options in the drop-down should also be labeled in a way that allows screen reader users to select a locale for their timezone.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple
